### PR TITLE
Protobuf attributes & missing types

### PR DESCRIPTION
**Pull request by DonBraulio**
This PR is built on top of #91 (check only latest commit for diff) and adds some attribute and feature types that were missing or recently added:
* Added support and test for feature dtypes: `BOOLEAN` and `STRING`. Serialization for these types and also `FLOAT32` and `INT32` is tested by changing the output feature dtypes of operators `OpI102`, etc. Now all dtypes are covered.
* Added support and/or tests for attribute types: `BOOLEAN`,  `REPEATED_STRING` and `MAP_STR_STR`. Serialization of these attributes is done in a new case `test_serialize_attributes`.

### CODE REVIEW

File: temporian/proto/core.proto [hunk 1]
```diff
@@ -71,6 +71,16 @@ message Operator {
       int64 integer_64 = 2;
       string str = 3;
       double float_64 = 4;
+      ListString list_str = 5;
```
Comment (getchebarne) [inline temporian/proto/core.proto:74]: I think we can use `repeated string` here

File: temporian/core/operators/base.py [hunk 2]
```diff
@@ -135,6 +135,16 @@ def check(self) -> None:
                 if available_output not in [v.key for v in definition.outputs]:
                     raise ValueError(f'Unexpected output "{available_output}".')
 
+            # Check that expected attributes are present
+            for expected_attr in definition.attributes:
```
Comment (achoum) [inline temporian/core/operators/base.py:139]: If possible, can we also check the type?

File: temporian/core/test/utils.py [hunk 3]
```diff
@@ -200,3 +201,80 @@ def __init__(self, event: Event):
...
+                pb.OperatorDef.Input(key="event"),
+            ],
+            outputs=[
+                pb.OperatorDef.Output(key="output"),
+            ],
+            attributes=[
+                pb.OperatorDef.Attribute(
+                    key="attr_int",
+                    type=pb.OperatorDef.Attribute.Type.INTEGER_64,
+                    is_optional=False,
```
Comment (achoum) [inline temporian/core/test/utils.py:221]: Nit: is_optional is optional. The default value is False.

**Review by achoum — APPROVED** [review]
That's great :)

File: temporian/core/test/utils.py (hunk 3 above)
Comment (DonBraulio) [reply temporian/core/test/utils.py:221]: That's a self explanatory var name 😆

**Pull request merged**